backend/views/views.py

```diff
@@ -34,12 +36,43 @@ def admin():
     return render_template('admin_page.html')
 
 
-@app.route('/userpage')
+@app.route('/userpage', methods=['GET'])
 @admin_permissions
 def user_page():
+    form = SearchForm(request.form)
     users = db.session.query(User, Role).filter(
         User.role_id == Role.id).order_by(User.id).all()
-    return render_template('user_page.html', users=users)
+    if request.args.get('search') != "" and request.args.get('search') >= MIN_SEARCH_STR:
+    # if form.validate_on_submit():
+        key = int(request.args.get('field_by'))
+        search_string = str(request.args.get('search'))
+        search_paremeters = []
+        search_users = []
+        for one_string in search_string.split(' '):
+            if len(one_string) >= MIN_SEARCH_STR:
+                search_paremeters.append(''.join(["%", one_string, "%"]))
+        for search_paremeter in search_paremeters:
+            conditions = {1:"users.name LIKE  '%s'" %(search_paremeter),
+                          2:"users.alias LIKE  '%s'" %(search_paremeter),
+                          3:"users.email LIKE  '%s'" %(search_paremeter)}
+            conditions[4] = ' '.join([conditions[1], "OR", conditions[2]])
+            conditions[5] = ' '.join([conditions[2], "OR", conditions[3]])
+            conditions[6] = ' '.join([conditions[1], "OR", conditions[3]])
+            conditions[7] = ' '.join([conditions[1], "OR", conditions[2], "OR", conditions[3]])
+            condition = conditions.get(key)
+            results = db.session.query(User, Role).filter(and_(
+                User.role_id == Role.id, str(condition))).distinct(User.id).order_by(User.id).all()
```

Comment:
You performs here one query to DB for each part of search string, which adds load to DB. Try to combine all filter expressions in one and perform one query.